nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExtensionCallMarkLogic.java

```diff
@@ -254,17 +259,21 @@ protected ExtensionResourceManager(DatabaseClient client, String resourceName) {
             client.init(resourceName, this);
         }
 
-        protected ServiceResultIterator callService(String method, AbstractWriteHandle writeHandle, RequestParameters parameters) {
+        protected ServiceResultIterator callService(String method, BytesHandle requestBody, RequestParameters parameters) {
             ServiceResultIterator serviceResultIterator;
             switch (method) {
                 case MethodTypes.GET_STR:
                     serviceResultIterator = getServices().get(parameters);
                     break;
                 case MethodTypes.POST_STR:
-                    serviceResultIterator = getServices().post(parameters, writeHandle);
+                    serviceResultIterator = getServices().post(parameters, requestBody);
                     break;
                 case MethodTypes.PUT_STR:
-                    serviceResultIterator = getServices().put(parameters, writeHandle, null);
+                    // Java Client requires a non-null, non-empty input for PUT calls
+                    if (requestBody.get() == null || requestBody.get().length == 0) {
+                        requestBody.set(new byte[]{});
```

Comment:
Isn't this a zero-length byte array? Does that mean ` || requestBody.get().length == 0` could be removed from the if condition since the outcome should be the same between an existing zero-length byte array and a new zero-length byte array?

Reply:
Good point, I'll test that out.

Reply:
Updated and added a test for an empty byte array